iFixit/react-commerce
# Add Strapi components for header

## Changelog

- Adds submenu menu component
- Adds header slot in store content type
- Adds seed data for header
- Minor frontend adaptations to account for content type changes

## QA

> 👀   This PR contains changes to Strapi config. Since the new Strapi staging server is not ready yet, we need to test this locally. The current Strapi server will be manually updated when this PR merged

1. Clone this repo
2. Make sure you are running on node v14 (if you are using nvm run `nvm use` in the project working directory)
3. Install dependencies `npm install`
4. Rename `backend/.env.example` to `backend/.env` (this will enable db seed)
5. Run dev server: `npm run dev`
6. Verify that the seed script runs correctly. If it does you should see the following messages on your terminal:
    <img width="833" alt="Screenshot 2021-12-01 at 12 03 06" src="https://user-images.githubusercontent.com/4640135/144223283-c0f44452-fa50-4999-a66a-0b9f35586266.png"> 
7. Log into local Strapi instance (http://localhost:1337/admin) with email `strapi@ifixit.com` and password `Password1`
8. Verify that a US store exists
9. Verify that US store has the header menu field set to "US Main"
10. Verify that menu "US Main" contains 3 submenus
11. Visit Vercel preview a verify that collection page still works
